src/python/pants/option/ranked_value.py

```diff
@@ -1,13 +1,40 @@
 # Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+from dataclasses import dataclass
 from enum import Enum
-from typing import Any, Dict, Iterator, List, Optional, Union, cast
+from functools import total_ordering
+from typing import Any, Dict, Iterator, List, Union
+
+
+@total_ordering
```

Comment:
With `@total_ordering`, you should also define `__eq__`: https://docs.python.org/3/library/functools.html#functools.total_ordering

Are you relying on the superclass's implementation? If so, consider adding this:

```python
def __eq__(other):
  return super().__eq__(other)
```

While repetitive, it does make it much more explicit that you're meeting the two requirements of `@total_ordering`.

Reply:
I think @dataclass is explicit enough, don't you?

Reply:
I don't follow. You're not using `@dataclass` here and are using `Enum`.

If we were using dataclass, the solution would be `@dataclass(ordering=True)`.

Reply:
Oops, yeah. But the exact same point applies to an Enum more strongly. By the very nature of a sealed immutable type equal, etc must work right. This is a language independent observation, you don't even need to know Python to assume it.

Reply:
Agreed, I don't see the value in explicit redundant boilerplate.